src/python/pants/option/options_fingerprinter.py

```diff
@@ -52,13 +52,10 @@ def combined_options_fingerprint_for_scope(cls, scope, options,
     hasher = sha1()
     pairs = options.get_fingerprintable_for_scope(scope, **kwargs)
     for (option_type, option_value) in pairs:
-      hasher.update(
-        # N.B. `OptionsFingerprinter.fingerprint()` can return `None`,
-        # so we always cast to bytes here.
-        six.binary_type(
-          fingerprinter.fingerprint(option_type, option_value)
-        )
-      )
+      fingerprint = fingerprinter.fingerprint(option_type, option_value)
+      if fingerprint is None:
+        fingerprint = 'None'
```

Comment:
I realise this is strange, but `six.binary_type(None) == 'None'` so this is behaviour preserving. Maybe not correct, but behaviour preserving :)

Reply:
Worth noting as a comment?

Reply:
Done